restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java

```diff
@@ -173,6 +175,72 @@ public Response postDoc(
         });
   }
 
+  @POST
+  @ManagedAsync
+  @ApiOperation(
+      value = "Write multiple documents in one request",
+      notes =
+          "Auto-generates an ID for the newly created document if an idPath is not provided as a query parameter. When an idPath is provided, this operation is idempotent.",
+      code = 201)
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 201, message = "Created", response = WriteDocResponse.class),
+        @ApiResponse(code = 400, message = "Bad request", response = Error.class),
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 403, message = "Forbidden", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal Server Error", response = Error.class)
+      })
+  @Path("collections/{collection-id}/batch")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response writeManyDocs(
+      @Context HttpHeaders headers,
+      @Context UriInfo ui,
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam("X-Cassandra-Token")
+          String authToken,
+      @ApiParam(value = "the namespace that the collection is in", required = true)
+          @PathParam("namespace-id")
+          String namespace,
+      @ApiParam(value = "the name of the collection", required = true) @PathParam("collection-id")
+          String collection,
+      @ApiParam(
+              value = "A JSON Lines payload where each line is a document to write",
+              required = true)
+          InputStream payload,
```

Comment:
can we add a `@NonNull` now when we have everything validated..

Reply:
What's the error response from dropwizard look like for that annotation?

Reply:
it's generated in `ViolationExceptionMapper`, but it requires a message, please do:

`@NotNull(message = "payload must not be null")`